editor/src/clj/editor/pipeline/tex_gen.clj

```diff
@@ -114,12 +116,21 @@
     (TextureUtil/createCombinedTextureImage (into-array texture-generator-results) texture-type)))
 
 (defn assemble-cubemap-texture-images
-  ^Graphics$TextureImage [side->texture-image]
+  ^TextureGenerator$GenerateResult [side->texture-image]
   (let [texture-images (into-array ((juxt :px :nx :py :ny :pz :nz) side->texture-image))
         type Graphics$TextureImage$Type/TYPE_CUBEMAP]
     (TextureUtil/createCombinedTextureImage texture-images type)))
 
 (defn make-preview-cubemap-texture-images
-  ^Graphics$TextureImage [images texture-profile]
+  ^TextureGenerator$GenerateResult [images texture-profile]
   (let [preview-profile (make-preview-profile texture-profile)]
     (make-cubemap-texture-images images preview-profile false)))
+
+(defn build-texture-resource-fn [resource user-data]
+  (let [digest-output-stream
+        (-> resource
+            (io/output-stream)
+            (digest/make-digest-output-stream "SHA-1"))]
+    ;; writeGenerateResultToOutputStream flushes the stream
+    (TextureUtil/writeGenerateResultToOutputStream (:texture-generator-result user-data) digest-output-stream)
+    (digest/completed-stream->hex digest-output-stream)))
```

Comment:
Do we need to flush the stream before calling `completed-stream->hex` here? 🤔

If `writeGenerateResultToOutputStream` flushes it internally, we should probably add a comment here.